RecipeApp.Core/RecipeConverter.cs

```diff
@@ -0,0 +1,270 @@
+                var tempUnit = Assembly.GetExecutingAssembly().CreateInstance(baseUnitType.FullName) as BaseUnit;
+                var tempUnitName = baseUnitType.GetProperties().FirstOrDefault(f =>f.Name == "Name")?.GetValue(tempUnit).ToString() ?? "";
+                if ( tempUnitName == unitName || tempUnitName+ "s" == unitName)
+                {
+                    unitType = baseUnitType;
+                    break;
+                }
+            }
+            if (unitType is null) return null;
+
+            BaseUnit unit = null;
+            if (ParseDouble(words[idx - 1], out double amount))
+            {
+                unit = Assembly.GetExecutingAssembly().CreateInstance(unitType.FullName) as BaseUnit;
+                unit.Amount = amount;
+            }
+
+            return unit;
+        }
+
+        private static BaseUnit SimplifyUnit(BaseUnit unit)
+        {
+            var otherUnits = Conversion.CookingUnits.Keys.Where(k => !k.Equals(unit.Name));
+
+            foreach (var newUnitName in otherUnits)
+            {
+                Type unitType = null;
+                foreach (var baseUnitType in Assembly.GetExecutingAssembly().GetTypes().Where(t => (t.BaseType?.Name ?? "") == "BaseUnit"))
+                {
+                    var tempUnit = Assembly.GetExecutingAssembly().CreateInstance(baseUnitType.FullName) as BaseUnit;
+                    var tempUnitName = baseUnitType.GetProperties().FirstOrDefault(f => f.Name == "Name")?.GetValue(tempUnit).ToString() ?? "";
+                    if (tempUnitName == newUnitName || tempUnitName + "s" == newUnitName)
+                    {
+                        unitType = baseUnitType;
+                        break;
+                    }
+                }
+
+                if (unitType is null) continue;
+
+                var newUnit = unit.ConvertTo(unitType);
+
+                if (newUnit.Amount > 1 && newUnit.Amount <= unit.Amount)
+                {
+                    unit = newUnit;
+                }
+            }
+
+            return unit;
+        }
+
+        private static void StandardiseCookingUnit(int index, string[] words)
+        {
+            var cookingUnit = words[index];
+            var multiplier = FindSiMultiplier(cookingUnit);
+            if (multiplier == -1) return;
+
+            var amountText = words[index - 1].Trim();
+
+            var isNumber = ParseDouble(amountText, out var amount);
+
+            if (!isNumber)
+            {
+                throw new InvalidRecipeException($"{amountText} is not a number.");
+            }
+
+            var amountMl = amount * multiplier;
+
+            if (amountMl > 100)
+            {
+                words[index] = "l";
+                words[index - 1] = (amountMl / 100).ToString("F2", CultureInfo.InvariantCulture);
+                return;
+            }
+
+            words[index] = "ml";
+            words[index - 1] = amountMl.ToString("F2", CultureInfo.InvariantCulture);
```

Comment:
Converting a number to a consistent format should go to a static function of its own, so that we consistently apply the same format throughout.

Reply:
extracted a method